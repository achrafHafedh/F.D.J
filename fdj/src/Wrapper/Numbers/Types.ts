export type NumbersProps = {
  dispatch: Function;
};
