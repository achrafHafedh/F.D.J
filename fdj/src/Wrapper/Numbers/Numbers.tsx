import { FunctionComponent } from "react";
import { NumbersProps } from "./Types";
import Number from "../../Components/Number";
import { NUMBERS_LENGTH } from "../../Config";

import "./Numbers.scss";

const Numbers: FunctionComponent<NumbersProps> = ({ dispatch }) => {
  const renderNumbers = () => {
    const listNumber = [];
    for (let index = 0; index < NUMBERS_LENGTH; index++) {
      listNumber.push(
        <Number key={index} dispatch={dispatch} className={index < 9 ? "small" : ""} index={index} value={index + 1} />
      );
    }
    return listNumber;
  };

  return <div className="container-numbers">{renderNumbers()}</div>;
};

export default Numbers;
