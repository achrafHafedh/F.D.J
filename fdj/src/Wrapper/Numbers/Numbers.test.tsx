import { render } from "@testing-library/react";

import Numbers from "./Numbers";

describe("<Numbers/>", () => {
  const props = {
    dispatch: () => "",
  };
  it("should render Numbers component", () => {
    const { container } = render(<Numbers {...props} />);
    expect(container).toMatchSnapshot();
  });
});
