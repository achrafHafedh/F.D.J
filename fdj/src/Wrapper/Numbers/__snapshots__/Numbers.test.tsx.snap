// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Numbers/> should render Numbers component 1`] = `
<div>
  <div
    class="container-numbers"
  >
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-0"
        type="checkbox"
        value="1"
      />
      <label
        class="small"
        for="number-0"
      >
        1
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-1"
        type="checkbox"
        value="2"
      />
      <label
        class="small"
        for="number-1"
      >
        2
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-2"
        type="checkbox"
        value="3"
      />
      <label
        class="small"
        for="number-2"
      >
        3
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-3"
        type="checkbox"
        value="4"
      />
      <label
        class="small"
        for="number-3"
      >
        4
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-4"
        type="checkbox"
        value="5"
      />
      <label
        class="small"
        for="number-4"
      >
        5
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-5"
        type="checkbox"
        value="6"
      />
      <label
        class="small"
        for="number-5"
      >
        6
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-6"
        type="checkbox"
        value="7"
      />
      <label
        class="small"
        for="number-6"
      >
        7
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-7"
        type="checkbox"
        value="8"
      />
      <label
        class="small"
        for="number-7"
      >
        8
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-8"
        type="checkbox"
        value="9"
      />
      <label
        class="small"
        for="number-8"
      >
        9
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-9"
        type="checkbox"
        value="10"
      />
      <label
        class=""
        for="number-9"
      >
        10
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-10"
        type="checkbox"
        value="11"
      />
      <label
        class=""
        for="number-10"
      >
        11
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-11"
        type="checkbox"
        value="12"
      />
      <label
        class=""
        for="number-11"
      >
        12
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-12"
        type="checkbox"
        value="13"
      />
      <label
        class=""
        for="number-12"
      >
        13
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-13"
        type="checkbox"
        value="14"
      />
      <label
        class=""
        for="number-13"
      >
        14
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-14"
        type="checkbox"
        value="15"
      />
      <label
        class=""
        for="number-14"
      >
        15
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-15"
        type="checkbox"
        value="16"
      />
      <label
        class=""
        for="number-15"
      >
        16
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-16"
        type="checkbox"
        value="17"
      />
      <label
        class=""
        for="number-16"
      >
        17
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-17"
        type="checkbox"
        value="18"
      />
      <label
        class=""
        for="number-17"
      >
        18
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-18"
        type="checkbox"
        value="19"
      />
      <label
        class=""
        for="number-18"
      >
        19
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-19"
        type="checkbox"
        value="20"
      />
      <label
        class=""
        for="number-19"
      >
        20
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-20"
        type="checkbox"
        value="21"
      />
      <label
        class=""
        for="number-20"
      >
        21
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-21"
        type="checkbox"
        value="22"
      />
      <label
        class=""
        for="number-21"
      >
        22
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-22"
        type="checkbox"
        value="23"
      />
      <label
        class=""
        for="number-22"
      >
        23
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-23"
        type="checkbox"
        value="24"
      />
      <label
        class=""
        for="number-23"
      >
        24
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-24"
        type="checkbox"
        value="25"
      />
      <label
        class=""
        for="number-24"
      >
        25
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-25"
        type="checkbox"
        value="26"
      />
      <label
        class=""
        for="number-25"
      >
        26
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-26"
        type="checkbox"
        value="27"
      />
      <label
        class=""
        for="number-26"
      >
        27
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-27"
        type="checkbox"
        value="28"
      />
      <label
        class=""
        for="number-27"
      >
        28
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-28"
        type="checkbox"
        value="29"
      />
      <label
        class=""
        for="number-28"
      >
        29
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-29"
        type="checkbox"
        value="30"
      />
      <label
        class=""
        for="number-29"
      >
        30
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-30"
        type="checkbox"
        value="31"
      />
      <label
        class=""
        for="number-30"
      >
        31
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-31"
        type="checkbox"
        value="32"
      />
      <label
        class=""
        for="number-31"
      >
        32
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-32"
        type="checkbox"
        value="33"
      />
      <label
        class=""
        for="number-32"
      >
        33
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-33"
        type="checkbox"
        value="34"
      />
      <label
        class=""
        for="number-33"
      >
        34
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-34"
        type="checkbox"
        value="35"
      />
      <label
        class=""
        for="number-34"
      >
        35
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-35"
        type="checkbox"
        value="36"
      />
      <label
        class=""
        for="number-35"
      >
        36
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-36"
        type="checkbox"
        value="37"
      />
      <label
        class=""
        for="number-36"
      >
        37
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-37"
        type="checkbox"
        value="38"
      />
      <label
        class=""
        for="number-37"
      >
        38
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-38"
        type="checkbox"
        value="39"
      />
      <label
        class=""
        for="number-38"
      >
        39
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-39"
        type="checkbox"
        value="40"
      />
      <label
        class=""
        for="number-39"
      >
        40
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-40"
        type="checkbox"
        value="41"
      />
      <label
        class=""
        for="number-40"
      >
        41
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-41"
        type="checkbox"
        value="42"
      />
      <label
        class=""
        for="number-41"
      >
        42
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-42"
        type="checkbox"
        value="43"
      />
      <label
        class=""
        for="number-42"
      >
        43
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-43"
        type="checkbox"
        value="44"
      />
      <label
        class=""
        for="number-43"
      >
        44
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-44"
        type="checkbox"
        value="45"
      />
      <label
        class=""
        for="number-44"
      >
        45
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-45"
        type="checkbox"
        value="46"
      />
      <label
        class=""
        for="number-45"
      >
        46
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-46"
        type="checkbox"
        value="47"
      />
      <label
        class=""
        for="number-46"
      >
        47
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-47"
        type="checkbox"
        value="48"
      />
      <label
        class=""
        for="number-47"
      >
        48
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-48"
        type="checkbox"
        value="49"
      />
      <label
        class=""
        for="number-48"
      >
        49
      </label>
      <div
        class="number"
      />
    </div>
    <div
      class="container-number"
    >
      <input
        class="visually-hidden"
        id="number-49"
        type="checkbox"
        value="50"
      />
      <label
        class=""
        for="number-49"
      >
        50
      </label>
      <div
        class="number"
      />
    </div>
  </div>
</div>
`;
